embassy-nrf/src/pwm.rs

```diff
@@ -376,20 +418,36 @@ pub struct SequenceConfig {
 impl Default for SequenceConfig {
     fn default() -> SequenceConfig {
         SequenceConfig {
-            counter_mode: CounterMode::Up,
-            max_duty: 1000,
-            prescaler: Prescaler::Div16,
-            sequence_load: SequenceLoad::Common,
             refresh: 0,
             end_delay: 0,
         }
     }
 }
 
+/// A composition of a sequence buffer and its configuration.
+#[non_exhaustive]
+pub struct Sequence<'d> {
+    /// The words comprising the sequence. Must not exceed 32767 words.
+    pub words: &'d mut [u16],
+    /// Configuration associated with the sequence.
+    pub config: SequenceConfig,
+}
+
+impl<'d> Sequence<'d> {
+    pub fn new(words: &'d mut [u16], config: SequenceConfig) -> Self {
+        Self { words, config }
+    }
+}
+
 /// How many times to run the sequence
 #[derive(Debug, Eq, PartialEq, Clone, Copy)]
 pub enum SequenceMode {
-    /// Run sequence n Times total
+    /// Run sequence n Times total.
+    /// 1 = Run sequence 0 once
+    /// 2 = Run sequence 0 and then sequence 1
+    /// 3 to 4 = Run sequence 0, sequence 1, sequence 0 and then sequence 1
```

Comment:
Times is a pretty bad way of describing all these options
Times(1) runs seq 1
Times(2) runs seq 0 then seq1
times(3) runs seq1 then seq0 then seq1

Times was a good name for my api, because mine was looping the same seq n times

Reply:
I will correct.

I don't believe the looping behaviour has changed.

Reply:
> Times(1) runs seq 1 

`Times(1)` has always run sequence 0. https://github.com/embassy-rs/embassy/blob/master/embassy-nrf/src/pwm.rs#L180

> Times(2) runs seq 0 then seq1

This is what I wrote.

> times(3) runs seq1 then seq0 then seq1

I will correct accordingly.